### ISSUE TITLE
deploymentTarget doesn't work as intended with Posit Cloud

### ISSUE THREAD

**Issue by mbaynton** [rstudio/rsconnect#808]
When deploying a new application (or whenever there is no dcf file strongly identifying an existing deployment target), rsconnect [attempts to identify an existing application on the server whose name suggests it might be the same as the application](https://github.com/rstudio/rsconnect/blob/9c427bbe61e96bcdcc257bbdbb4444349ec9897b/R/deploymentTarget.R#L29-L30).

This does not work on posit cloud because:
1. Posit cloud's application names are essentially random and so will never match an application name supplied by the user or inferred by rsconnect
2. The cloud client currently [filters to applications of type `connect`](https://github.com/rstudio/rsconnect/blob/9c427bbe61e96bcdcc257bbdbb4444349ec9897b/R/client-cloud.R#L61-L62), which exclude `static` applications.

The result is that we make a potentially expensive list api call to filter applications by name in memory with a virtual guarantee none are going to match. We should probably reimagine how this feature works for Posit Cloud, or skip the api call entirely.

**Comment by hadley** [comment]
Is there any chance of using meaningful `appName`s on cloud? That would make the behaviour align more closely with both connect and shinyapps.io.

**Comment by mbaynton** [comment]
Yeah, our meaningful names are on a different entity that cloud calls `content`. Fixing this in a way that preserves the intent of the feature would involve consulting our content entities.

**Comment by kippandrew** [comment]
For context, on Posit Cloud, applications are not a subtype of _Content_, but rather they are an entity that represents a servable resource with a unique URL. For example, _Project_ is a subtype of _Content_ and may have one or more applications associated with it (each accessible via its own URL). A Shiny application deployed to Posit Cloud is represented as another subtype of _Content_ called an _Output_. An _Output_ may also have one or more _Application_ entities associated with depending on the nature of the output (static vs. dynamic). In both cases, unlike Posit Connect, an _Application_ is not a subtype of _Content_. 

The bottom line is that the Posit Cloud domain model has diverged in some meaningful ways from shinyapps.io and Posit Connect. We might consider re-examining this tool's assumptions rather than attempting to align the domain models across all the products. We could also consider working to define a publishing contract/API that is abstracted enough to accommodate the differences between the products. Another solution might be to create a separate tool for publishing to Posit Cloud.
